syncopy/datatype/discrete_data.py

```diff
@@ -51,8 +51,15 @@ def data(self):
 
     @data.setter
     def data(self, inData):
+        """ Also checks for integer type of data """
+        # this comes from BaseData
         self._set_dataset_property(inData, "data")
 
+        if inData is not None:
+            # probably not the most elegant way..
+            if 'int' not in str(self.data.dtype):
```

Comment:
I think this would be the "right" way to do it 
`if np.issubdtype(self.data.dtype, np.integer)`

Reply:
uuh, yes that looks much better!